composeml/label_times/object.py

```diff
@@ -17,65 +17,84 @@ def __init__(
         self,
         data=None,
         target_entity=None,
-        name=None,
-        label_type=None,
+        target_types=None,
+        target_columns=None,
         search_settings=None,
         transforms=None,
         *args,
         **kwargs,
     ):
         super().__init__(data=data, *args, **kwargs)
         self.target_entity = target_entity
-        self.label_name = name
-        self.label_type = label_type
+        self.target_columns = target_columns or []
+        self.target_types = target_types or {}
         self.search_settings = search_settings or {}
         self.transforms = transforms or []
         self.plot = LabelPlots(self)
 
-        if not self.empty:
-            self._check_label_name()
-            self._check_label_type()
+        if not self.empty: self._check_label_times()
+
+    def _assert_single_target(self):
+        info = 'must first select an individual target'
+        assert self._is_single_target, info
 
-    def _check_label_name(self):
+    def _check_label_times(self):
         """Checks whether the target exists in the data frame."""
-        if self.label_name is None:
-            self.label_name = self._infer_label_name()
+        if not self.target_columns:
+            self.target_columns = self._infer_target_columns()
+
+        missing = pd.Index(self.target_columns).difference(self.columns)
+        info = 'target variable(s) not found: %s'
+        assert missing.empty, info % missing.tolist()
+
+        if not isinstance(self.target_types, pd.Series):
+            self.target_types = pd.Series(self.target_types)
 
-        info = 'target variable not found: %s' % self.label_name
-        assert self.label_name in self.columns, info
+        if self.target_types.empty:
+            self.target_types = self._infer_target_types()
 
-    def _infer_label_name(self):
+    def _infer_target_columns(self):
```

Comment:
Maybe update the docstring for this as now it is possible to return multiple target values instead of just one, but the docstring only references a single value.

Also, is it possible to have a situation where no target columns are found? If so, we should check for it and raise an error/warning depending on the downstream implications.

Reply:
I think an edge case would be constructing a label times object that only contains a column for the target entity or time. I added an assertion to check for this situation.